crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java

```diff
@@ -239,11 +256,9 @@ private T internalFromImpl(TypeDef definition, Set<String> visited, List<Interna
         continue;
       }
 
-      final PropertyFacade facade = new PropertyFacade(property, accessors, currentSchemaSwaps);
+      ClassRef potentialSchemaSwap = schemaSwaps.lookupAndMark(definition.toReference(), name).orElse(null);
```

Comment:
If I correctly follow this change, here you cannot mark just yet the SchemaSwap as used, as it will mark the annotation to be "used" even in this condition:
 - Matched `Class`
 - The Class doesn't have the specified `Property` (e.g. a matching field)
 
 This operation should be performed only after `process` has been called on the `PropertyFacade` (and the `SchemaSwap` has been matched).

Reply:
schemaSwapt.lookupAndMark  accepts two arguments, the second being the property name. (Line 253: `String name = property.getName();`)
So I'm not sure I understand your concert?
Btw, I thought the case of "matched class, unmatched field" is being tested in JsonSchemaTest.shouldThrowIfSchemaSwapHasUnmatchedField. This test is still passing and I only extended it with assert on the error message. But the error is still thrown.

Reply:
Of is your concern that the `name` could be overriden by `@JsonProperty` & friends? IMHO it makes more sense for the `fieldName` parameter to reference the fields as is defined in Java, but that's just my opinion.

Reply:
> it makes more sense for the fieldName parameter to reference the fields as is defined in Java, but that's just my opinion.

This makes sense, but, using `name` might not be the name extracted by the `PropertyFacade` constructor:
https://github.com/fabric8io/kubernetes-client/blob/808084e0e348d5b377ac31fd89a979996867d93e/crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java#L414-L426

in some cases, the `PropertyFacade` is extracting the field name from constructor/getter/setter 
and this implementation will not respect the extracted name as opposed to the underlying field name.
We can either refactor the logic and use the `PropertyFacade` name extraction logic everywhere or simply document this different behavior.

Reply:
I see, and the only way how that can happen is a `@JsonProperty` on the field itself or getter (get* or is*) or setter. 
I can only say that to me, as the user of the library, would never occur that `fieldName` could reference the name after transformations. 
Maybe it's because the word "field" is used, whose meaning in Java is well defined and is different from the term "property". (Field is a class variable, property is a getter/setter pair - which often is backed by a field). Also the annotation is `@JsonProperty`, not "JsonField".
Or maybe it's because the reason I am dealing with `@SchemaSwaps` in the first place is that I cannot place annotations (like `@JsonProperty`) on the field directly.
So with your permission, I will add Javadocs on `@SchemaSwap`, describing the currently implemented behaviour.

Reply:
To me, this "limitation" is totally fine, please just make it explicit in this doc:
https://github.com/fabric8io/kubernetes-client/blob/master/doc/CRD-generator.md#iofabric8crdgeneratorannotationschemaswap

mentioning something along the lines:
> the name of the field is restricted to the original `fieldName` and should be backed by a matching concrete field of the matching class. Getters, setters, and constructors are not taken into consideration.

cc. @metacosm to check if he is on board with this decision.

Reply:
I think that should be fine but this should be documented as a breaking change, though…

Reply:
I agree that it "might" be breaking, but, at the same time, this PR is solving some fundamental issues with the implementation and I doubt anyone else is using it just yet.

Reply:
Maybe we can add a line to the Changelog for reference, wdyt?